src/server/game/Spells/Auras/SpellAuras.cpp

```diff
@@ -2044,37 +2047,30 @@ void Aura::TriggerProcOnEvent(uint8 procEffectMask, AuraApplication* aurApp, Pro
 
 void Aura::HeartbeatResistance(uint32 diff, Unit* caster)
 {
-    m_heartBeatTimer += diff;
-
-    if (m_heartBeatTimer < IN_MILLISECONDS)
-        return;
-
     Unit* target = GetOwner()->ToUnit();
 
     if (!target || !caster)
         return;
 
-    if (target->GetTypeId() == TYPEID_UNIT && caster->GetTypeId() == TYPEID_PLAYER)
+    if (m_heartBeatTimer <= diff)
     {
-        m_heartBeatTimer = (uint32)GetApplyTime() - m_heartBeatTimer;
-
-        if (m_heartBeatTimer == (uint32)std::floor(GetMaxDuration() * 0.25f / IN_MILLISECONDS + 0.5f) ||
-            m_heartBeatTimer == (uint32)std::floor(GetMaxDuration() * 0.50f / IN_MILLISECONDS + 0.5f) ||
-            m_heartBeatTimer == (uint32)std::floor(GetMaxDuration() * 0.75f / IN_MILLISECONDS + 0.5f))
+        if (target->GetTypeId() == TYPEID_UNIT && caster->GetTypeId() == TYPEID_PLAYER)
         {
-            SpellSchoolMask schoolMask = m_spellInfo->GetSchoolMask();           
-            uint32 resistance = schoolMask != SPELL_SCHOOL_MASK_NORMAL ? target->GetResistance(GetFirstSchoolInMask(schoolMask)) : 0;       
-            uint32 breakPct = uint32(resistance / powf(target->getLevel(), 1.441f) * 0.10f * 100) + 5;
-            
+            SpellSchoolMask schoolMask = m_spellInfo->GetSchoolMask();
+            uint32 resistance = schoolMask != SPELL_SCHOOL_MASK_NORMAL ? target->GetResistance(GetFirstSchoolInMask(schoolMask)) : 0;
+            uint32 breakPct = uint32(resistance / std::powf(target->getLevel(), 1.441f) * 0.10f * 100) + 5;
+
             if (roll_chance_i(breakPct))
             {
                 Remove();
                 TC_LOG_DEBUG("spells", "Aura::HeartbeatResistance: Breaking creature aura %u. Seconds passed %u with chance %u.", m_spellInfo->Id, m_heartBeatTimer, breakPct);
             }
         }
+
+        m_heartBeatTimer = 5 * IN_MILLISECONDS;
```

Comment:
`+=`, and should be `duration/4`, 5000 was an example

Reply:
ok